File: Data Structures & Algorithms/validate-parentheses/submission-3.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack=[]
        bracket={"}":"{","]":"[",")":"("}
        for i in s:
            if i in bracket.values():
                stack.append(i)
            elif i in bracket.keys():
                if stack and stack[-1]==bracket[i]:
                    stack.pop()
                else:
                    return False
        return len(stack)==0

        

File: Data Structures & Algorithms/minimum-stack/submission-3.cpp
class MinStack {
private:
    stack<pair<int,int>> s;
   
public:
    MinStack() {
        
    }
    
    void push(int val) {
       if (s.empty()){
        s.push({val,val});
       } else {
        int m = min(val,s.top().second);
        s.push({val,m});
       }
    }
    
    void pop() {
        s.pop();
    }
    
    int top() {
        return s.top().first;
    }
    
    int getMin() {
        return s.top().second;
    }
};
